server/src/infra/repositories/search.repository.ts

```diff
@@ -172,6 +179,27 @@ export class SearchRepository implements ISearchRepository {
     }));
   }
 
+  @GenerateSql({ params: [DummyValue.STRING] })
+  async searchPlaces(placeName: string): Promise<GeodataPlacesEntity[]> {
+    return await this.geodataPlacesRepository
+      .createQueryBuilder('geoplaces')
+      .where(`f_unaccent(name) %>> f_unaccent(:placeName)`)
+      .orWhere(`f_unaccent("admin2Name") %>> f_unaccent(:placeName)`)
+      .orWhere(`f_unaccent("admin1Name") %>> f_unaccent(:placeName)`)
+      .orWhere(`f_unaccent("alternateNames") %>> f_unaccent(:placeName)`)
+      .orderBy(
+        `
+        COALESCE(f_unaccent(name) <->>> f_unaccent(:placeName), 0) +
+        COALESCE(f_unaccent("admin2Name") <->>> f_unaccent(:placeName), 0) +
+        COALESCE(f_unaccent("admin1Name") <->>> f_unaccent(:placeName), 0) +
+        COALESCE(f_unaccent("alternateNames") <->>> f_unaccent(:placeName), 0)
```

Comment:
I trust this just does some magic that makes it work lol

Reply:
Haha let me explain:

`f_unaccent` removes accents from the words to make searching easier. The index uses this, so only `placeName` has to be un-accented at search time. But Postgres won't use the index if the columns don't also have `f_unaccent`, so we add this function to the columns too.

The `%>>`  operator checks if the two strings are similar using strict word similarity, meaning it checks against word boundaries instead of comparing the entire strings.

The `<->>>` operator, like `%>>`, compares two strings with strict word similarity, but the difference is that it returns a score instead of a boolean. Since it returns null if one of the strings is null, `COALESCE` prevents that from affecting the other comparisons.

We take the sum of all of those to make the ranking more robust. If the right location doesn't show up in the dropdown, the user can type more details about the state or country and the sum ranking will make the right location rank higher. If it was a strict ordering by city, state etc., then it might never rise to the top.

Reply:
Who tf figured `<->>>` is a good, precise operator for string comparison? lmao